src/org.spec.ts

```diff
@@ -255,4 +255,365 @@ describe('org', () => {
+      });
+      expect(dbResult).not.toBeNull();
+    });
+
+    test('Malformed name should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        'INVALID_Name',
+        {
+          name: 'INVALID_Name',
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.MALFORMED_ORG_NAME);
+    });
+
+    test('Non matching name should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        ORG_NAME,
+        {
+          name: `a.${ORG_NAME}`,
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.INVALID_ORG_NAME);
+    });
+
+    test('Malformed Awala endpoint should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        ORG_NAME,
+        {
+          name: ORG_NAME,
+          memberAccessType: 'INVITE_ONLY',
+          awalaEndpoint: 'INVALID_AWALA_ENDPOINT',
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.MALFORMED_AWALA_ENDPOINT);
+    });
+
+    test('Record update errors should be propagated', async () => {
+      const connection = getConnection();
+      await connection.close();
+      const orgData: OrgSchema = {
+        name: ORG_NAME,
+        memberAccessType: 'INVITE_ONLY',
+      };
+
+      const error = await getPromiseRejection(
+        async () =>
+          updateOrg(ORG_NAME, orgData, {
+            dbConnection: connection,
+            logger: mockLogging.logger,
+          }),
+        Error,
+      );
+
+      expect(error).toHaveProperty('name', 'MongoNotConnectedError');
+    });
+
+    describe('getOrg', () => {
+      test('valid name should return the corresponding data', async () => {
+        const connection = getConnection();
+        const orgModel = getModelForClass(OrgModelSchema, {
+          existingConnection: connection,
+        });
+        await orgModel.create({
+          name: ORG_NAME,
+          memberAccessType: MemberAccessType.OPEN,
+          awalaEndpoint: AWALA_ENDPOINT,
+        });
+
+        const methodResponse = await getOrg(ORG_NAME, {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        });
+
+        requireSuccessfulResult(methodResponse);
+        expect(methodResponse.result).toMatchObject({
+          name: ORG_NAME,
+          memberAccessType: 'OPEN',
+          awalaEndpoint: AWALA_ENDPOINT,
+        });
+      });
+
+      test('invalid name should return NOT_FOUND error', async () => {
```

Comment:
Similarly, I'd say "invalid" -> "non-existing".

Also, `NOT_FOUND` suggests there's an identifier by that name, probably one for the `404` response. I'd avoid that... maybe something like `Non-existing name should be refused`?